lms/djangoapps/grades/api/views.py

```diff
@@ -128,28 +137,48 @@ def get(self, request, course_id):
         Return:
             A JSON serialized representation of the requesting user's current grade status.
         """
-        username = request.GET.get('username')
-
-        # only the student can access her own grade status info
-        if request.user.username != username:
-            log.info(
-                'User %s tried to access the grade for user %s.',
-                request.user.username,
-                username
-            )
-            return self.make_error_response(
-                status_code=status.HTTP_404_NOT_FOUND,
-                developer_message='The user requested does not match the logged in user.',
-                error_code='user_mismatch'
-            )
 
+        # Use the request user's if none provided.
+
+        if 'username' in request.GET:
+            username = request.GET.get('username')
+        else:
+            username = request.user.username
```

Comment:
This code can also move to the separate helper method for `_get_effective_user` - since it's only used later in this method for determining what you're currently calling `grade_user`.